### ISSUE TITLE
nightly-2021-09-03: Compiler hang in project with a lot of `axum` crate routes

### ISSUE THREAD

**Issue by paolobarbolini** [rust-lang/rust#89601]
https://github.com/paolobarbolini/rustc-pr85868-regression

This small reproduction takes 20 seconds to `cargo check` on 50171c310cd15e1b2d3723766ce64e2e4d6696fc, 45 seconds on 371f3cd3fe523d0b398ed1db1620667c53ba7d0.
On the real project, which is the one I run the bisection on, 50171c310cd15e1b2d3723766ce64e2e4d6696fc takes 1 minute, 371f3cd3fe523d0b398ed1db1620667c53ba7d0 hangs for 12+ hours.

## Bisection

searched nightlies: from nightly-2021-08-25 to nightly-2021-09-10
regressed nightly: nightly-2021-09-03
searched commit range: https://github.com/rust-lang/rust/compare/50171c310cd15e1b2d3723766ce64e2e4d6696fc...371f3cd3fe523d0b398ed1db1620667c53ba7d02
regressed commit: https://github.com/rust-lang/rust/commit/371f3cd3fe523d0b398ed1db1620667c53ba7d02

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.0</summary>

Host triple: x86_64-unknown-linux-gnu
Reproduce with:
```bash
cargo bisect-rustc 2021-08-25 --end 2021-09-10 --timeout 90 
```
</details>

---

Bisected with https://github.com/rust-lang/cargo-bisect-rustc/pull/135 :smiley:

**Comment by shirshak55** [comment]
@jplatte It's because they used a box to fix this issue. So I don't think the underlying problem is solved?

**Comment by davidpdrsn** [comment]
@shirshak55 Thats correct.

**Comment by jplatte** [comment]
I didn't want to imply that the underlying problem is solved. Just thought this info would be interesting for people who are subscribed to this issue because they want to use axum + a recent Rust version.

**Comment by pnkfelix** [comment]
closing as resolved, both due to axum-local workarounds first, and later due to #89195 being resolved by #90423
